breakpad-symbols/src/sym_file/parser.rs

```diff
@@ -3,19 +3,31 @@
 
 use nom::IResult::*;
 use nom::*;
-use range_map::{Range, RangeMap};
+use rangemap::RangeMap;
 use tracing::warn;
 
 use std::collections::HashMap;
 use std::fmt::Debug;
+use std::ops::Range;
 use std::str;
 use std::str::FromStr;
 
-use minidump_common::traits::IntoRangeMapSafe;
-
 use crate::sym_file::types::*;
 use crate::SymbolError;
 
+trait RangeIntersects {
+    fn intersects(&self, other: &Self) -> bool;
+}
+
+impl<T: Ord> RangeIntersects for Range<T> {
+    fn intersects(&self, other: &Self) -> bool {
+        self.start < self.end
+            && other.start < other.end
+            && self.start <= other.end
+            && other.start < self.end
```

Comment:
I think this merits a bit of a comment. The first two checks just seem to be "this isn't an empty (or inverted) range" but I'm a bit confused by why one of the comparisons is <=...

Reply:
You caught a bug, the <= is wrong. I was following [our rectangle intersection code](https://searchfox.org/mozilla-central/rev/0069a0cdc3cd136ea7411521219df6665f260be7/gfx/2d/BaseRect.h#105-107) and still screwed it up.